packages/ag-charts-website/src/features/codeSandbox/utils/codeSandbox.ts

```diff
@@ -0,0 +1,159 @@
+import type { InternalFramework } from '@ag-grid-types';
+import type { FileContents } from '@features/examples-generator/types';
+import { pathJoin } from '@utils/pathJoin';
+
+const CREATE_CODE_SANDBOX_URL = 'https://codesandbox.io/api/v1/sandboxes/define?json=1';
+const CODE_SANDBOX_URL_PREFIX = 'https://codesandbox.io/p/sandbox/';
+
+type SandboxFiles = Record<string, { content: string }>;
+
+const getCodeSandboxTemplate = (config: object) => {
+    return { ...config, tags: ['ag-grid', 'ag-charts', 'example'], published: false };
+};
+
+function isReactFramework(internalFramework: InternalFramework) {
+    return new Set(['reactFunctional', 'reactFunctionalTs']).has(internalFramework);
+}
+
+const getPathForFile = ({
+    fileName,
+    internalFramework,
+}: {
+    fileName: string;
+    internalFramework: InternalFramework;
+}) => {
+    if (!isReactFramework(internalFramework)) {
+        return fileName;
+    }
+
+    if (fileName === 'index.html') {
+        return `public/index.html`;
+    }
+
+    if (/([a-zA-Z0-9\\s_.])+(.js|.jsx|.tsx|.ts|.css)$/.test(fileName)) {
```

Comment:
What's the first bit doing? The [a-z] stuff?

Reply:
This was taken from grid, but I guess it's not necessary, since if it's src-looking, it should go in the `src` folder. Have removed that bit.